app/src/main/java/com/realityexpander/tasky/agenda_feature/data/repositories/eventRepository/eventRepositoryImpls/EventRepositoryImpl.kt

```diff
@@ -36,6 +38,24 @@ class EventRepositoryImpl(
         }
     }
 
+
+    // • UPSERT
+
+    override suspend fun upsertEventLocally(event: AgendaItem.Event): ResultUiText<Void> {
+        return try {
+            eventDao.upsertEvent(event.toEntity())  // save to local DB ONLY
+
+            ResultUiText.Success()
+        } catch (e: CancellationException) {
+            throw e
+        } catch (e: Exception) {
+            ResultUiText.Error(UiText.Str(e.message ?: "upsertEvent error"))
```

Comment:
What could go wrong here?

Reply:
Out of space on the device. Maybe some kind of configuration error? 

Is there a better way to handle this?

Reply:
I think I've never seen Room throw an exception in my life, but out of storage is a valid point, so please ignore this comment 👍